presto-hive/src/main/java/io/prestosql/plugin/hive/metastore/cache/CachingHiveMetastore.java

```diff
@@ -218,38 +230,33 @@ public PartitionStatistics load(WithIdentity<HivePartitionName> key)
                         return loadPartitionsByNames(partitionNames);
                     }
                 }, executor));
+        caches.add(partitionCache);
 
         tablePrivilegesCache = newCacheBuilder(expiresAfterWriteMillis, refreshMills, maximumSize, statsRecording)
                 .build(asyncReloading(CacheLoader.from(key -> loadTablePrivileges(key.getDatabase(), key.getTable(), key.getOwner(), key.getPrincipal())), executor));
+        caches.add(tablePrivilegesCache);
 
         rolesCache = newCacheBuilder(expiresAfterWriteMillis, refreshMills, maximumSize, statsRecording)
                 .build(asyncReloading(CacheLoader.from(this::loadRoles), executor));
+        caches.add(rolesCache);
 
         roleGrantsCache = newCacheBuilder(expiresAfterWriteMillis, refreshMills, maximumSize, statsRecording)
                 .build(asyncReloading(CacheLoader.from(this::loadRoleGrants), executor));
+        caches.add(roleGrantsCache);
 
         grantedPrincipalsCache = newCacheBuilder(expiresAfterWriteMillis, refreshMills, maximumSize, statsRecording)
                 .build(asyncReloading(CacheLoader.from(this::loadPrincipals), executor));
+        caches.add(grantedPrincipalsCache);
 
         configValuesCache = newCacheBuilder(expiresAfterWriteMillis, refreshMills, maximumSize, statsRecording)
                 .build(asyncReloading(CacheLoader.from(this::loadConfigValue), executor));
+        caches.add(configValuesCache);
     }
 
     @Managed
     public void flushCache()
     {
-        databaseNamesCache.invalidateAll();
-        tableNamesCache.invalidateAll();
-        viewNamesCache.invalidateAll();
-        partitionNamesCache.invalidateAll();
-        databaseCache.invalidateAll();
-        tableCache.invalidateAll();
-        partitionCache.invalidateAll();
-        partitionFilterCache.invalidateAll();
-        tablePrivilegesCache.invalidateAll();
-        tableStatisticsCache.invalidateAll();
-        partitionStatisticsCache.invalidateAll();
-        rolesCache.invalidateAll();
+        caches.forEach(Cache::invalidateAll);
```

Comment:
I don’t see why we need to duplicate the list of caches just to invalidate them. The old code seems fine. This change is unrelated to fixing grant/revoke, so please make a separate PR and we can discuss it independently.

Reply:
The old code was buggy, 4 caches weren't cleaned, so I added this to prevent such things in the future. Now the title of this pr will be different.

Reply:
now it is buggy in different way. Still someone might forget to add a cache to `caches`. Please undo the refactor or `please make a separate PR and we can discuss it`